### PR TITLE
build and release binaries (for updated guide on medium)

### PR DESCRIPTION
**Pull request by nicolasochem**

### CODE REVIEW

File: .github/workflows/rust-release.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - target: x86_64-pc-windows-gnu
+            archive: zip
+          - target: x86_64-unknown-linux-musl
+            archive: tar.gz
+          - target: x86_64-apple-darwin
+            archive: zip
```
Comment (itkach) [inline .github/workflows/rust-release.yml:18]: can we add apple silicon target?

File: .github/workflows/rust-release.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+      - uses: actions/checkout@master
```
Comment (itkach) [inline .github/workflows/rust-release.yml]: at master or at main?

Comment (nicolasochem) [reply .github/workflows/rust-release.yml]: made it v3

File: .github/workflows/rust-release.yml (hunk 1 above)
Comment (nicolasochem) [reply .github/workflows/rust-release.yml:18]: it looks unsupported with this action
https://github.com/rust-build/rust-build.action/issues/4#issuecomment-1044914831

**Review by itkach — APPROVED** [review]

**Pull request merged**